### DISCUSSION TITLE
apt warning: deb/dists/debian/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg)

### DISCUSSION THREAD

**Discussion by slrslr** [yggdrasil-network/yggdrasil-go#1177] [Q&A]
After upgrading Debian 11(oldstable) to Debian 12(stable), "sudo apt update" started complaining about:

> SW: http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb/dists/debian/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.

$ find /etc/apt -type f -iname \*yggd\*
```
/etc/apt/sources.list.d/yggdrasil.list
/etc/apt/sources.list.d/yggdrasil.list.save
```

$ grep -Ria yggd /etc/apt
```
/etc/apt/sources.list.d/yggdrasil.list:deb http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb/ debian yggdrasil
/etc/apt/sources.list.d/yggdrasil.list.save:deb http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb/ debian yggdrasil
```

$ sudo apt-key list
does not show anything that looks familiar and that i could remove (sudo apt-key del [KEYID]) and then try to obtain new key placing it in a correct location which may be /etc/apt/trusted.gpg.d/ ?

$ apt search yggdrasil
```
golang-github-arceliar-ironwood-dev/stable,stable 0.0~git20221115.ec61cea-1 all
  Routing library with public keys as addresses (library)

golang-github-yggdrasil-network-yggdrasil-go-dev/stable,stable 0.4.7-1 all
  Fully encrypted global-scale mesh network (library)

yggdrasil/unknown,now 0.5.8 amd64 [installed]
  Yggdrasil Network

yggdrasil-develop/unknown 0.5.8-16-g874083d amd64
  Yggdrasil Network

yggdrasil-vyos13/unknown 0.5.5 amd64
```

I see that https://yggdrasil-network.github.io/installation-linux-deb.html lists some Debian installation steps incl. key related.

Please what do you suggest to get rid of that apt warning and keep up to date?

**Comment by slrslr** [comment]
After upgrading Debian 12->13:

$ sudo apt modernize-sources # recommended [here](https://www.debian.org/releases/trixie/release-notes/upgrading.en.html#cleanup-after-the-upgrade)
$ sudo apt --audit update
```
Hit:11 http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb debian InRelease                     
Err:11 http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb debian InRelease
Warning: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. OpenPGP signature verification failed: http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb debian InRelease: Sub-process /usr/bin/sqv returned an error code (1), error message is: Missing key AA3470529B8826A86C2BC578BC1BF63BD10B8F1A, which is needed to verify signature.
Warning: Failed to fetch http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb/dists/debian/InRelease  Sub-process /usr/bin/sqv returned an error code (1), error message is: Missing key AA3470529B8826A86C2BC578BC1BF63BD10B8F1A, which is needed to verify signature.
Warning: Some index files failed to download. They have been ignored, or old ones used instead.
Notice: Missing Signed-By in the sources.list(5) entry for 'http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb'
```

New repository definition format: https://wiki.debian.org/SourcesList#Unstable

\$ cat /etc/apt/sources.list.d/yggdrasil.sources
```
Types: deb
URIs: http://neilalexander.s3.dualstack.eu-west-2.amazonaws.com/deb/
Suites: debian
Components: yggdrasil
Signed-By: 
```

$ apt list yggdrasil
```
yggdrasil/stable,now 0.5.12-2+b4 amd64 [installed]
yggdrasil/stable 0.5.12-2+b4 i386
```

$ apt info yggdrasil -a
[here](https://justpaste.it/gxmtu)

In the meantime, I have removed the repository definition, since official main repository has same yggdrasil package version:

\$ sudo gio trash /etc/apt/sources.list.d/*yggdrasil*

\$ sudo apt update

that removed the warnings shown earlier. Any fix for this please @neilalexander  ?

**Reply by oooo-ps** [reply]
Try to build Yggdrasil from the repo, just make sure your go version is up to date:
https://github.com/yggdrasil-network/yggdrasil-go?tab=readme-ov-file#building

**Reply by slrslr** [reply]
My problem is apt related. I do not want to spend time manually updating the apps.

**Reply by neilalexander** [reply]
I'm aware of the problem, I've just been too short on time recently to address it. I will try to look into it this week.
